File: src/dailyprogrammer/easy/_393/cs/MakingChange.cs
using System;

namespace MakingChange {
    class Program {
        enum Coin {
            FiveDollarCoin = 500, DollarCoin = 100,
            Quarter = 25, Dime = 10, Nickel = 5, Penny = 1
        }
        
        static void Main(String[] args) {
            Console.WriteLine("change(0) => {0}", Change(0));
            Console.WriteLine("change(12) => {0}", Change(12));
            Console.WriteLine("change(468) => {0}", Change(468));
            Console.WriteLine("change(123456) => {0}", Change(123456));
        }
        // Prints number of coins needed of each coin denomination
        // Returns total number of coins needed across all coin values
        static int Change(int money) {
            int yourChange = 0;
            int[] coinValues = (int[])Enum.GetValues(typeof(Coin));
            Array.Reverse(coinValues);
            foreach(int value in coinValues ) {
                int remainder = money % value;
                int amtCoins = money / value;
                yourChange += amtCoins;
                money = remainder;
                Console.Write("{0} {1}, ", amtCoins, 
                    Enum.GetName(typeof(Coin), value));
            }
            Console.WriteLine();
            return yourChange;
        }
    } // end class
} // end namespace

File: src/dailyprogrammer/easy/_393/MakingChange.c
#include <stdio.h>

// Sum all values in a int array
int sum(int any_list[], const int list_size) {
    int summed = 0;
    for(int i =0; i < list_size; i++) {
        summed += any_list[i];
    }
    return summed;
}

// Gives number of coins needed to make optimal change from money
int change(int money) {
    int coins[] = {500, 100, 25, 10, 5, 1};
    const int coins_len = sizeof coins / sizeof coins[0];
    int amt_coins[coins_len];
    for (int i = 0; i < coins_len; i++) {
        int coin = coins[i];
        int remainder = money % coin;
        amt_coins[i] = money / coin;
        money = remainder;
    }
    return sum(amt_coins, coins_len);
}

int main() {
    printf("change(0) => %i\n", change(0));
    printf("change(12) => %i\n", change(12));
    printf("change(468) => %i\n", change(468));
    printf("change(123456) => %i", change(123456));
    return 0;
}